api.ts

```diff
@@ -812,6 +809,60 @@ export interface SyncUpdateResult<K extends string, L extends string, SchemaT ex
  */
 export type GenericSyncUpdateResult = SyncUpdateResult<any, any, any>;
 
+/**
+ * Type definition for a single marshaled update result returned by a sync table update function.
+ * @hidden
+ */
+export type SyncUpdateSingleResultMarshaled<
+  K extends string,
+  L extends string,
+  SchemaT extends ObjectSchemaDefinition<K, L>,
+> = SyncUpdateSingleResultMarshaledSuccess<K, L, SchemaT> | SyncUpdateSingleResultMarshaledError;
+
+/**
+ * Type definition for a single marshaled update success result returned by a sync table update function.
+ * @hidden
+ */
+export interface SyncUpdateSingleResultMarshaledSuccess<
+  K extends string,
+  L extends string,
+  SchemaT extends ObjectSchemaDefinition<K, L>,
+> {
+  success: true;
```

Comment:
I don't have a strong opinions either way, but something like `updateOutcome: "Success" | "Error"` could offer more wiggle room in the future than `success: boolean`, and maybe a tiny bit of defense against someone doing `if (result.success)` and not noticing that `success` was undefined instead of false.

Reply:
This is all internal right now (post marshaling once we leave the pack-owned code), so probably doesn't really matter since we can change it in the future as needed.

Reply:
Even if we changed it, if it's returned by the lambda I think we'd have to support the old values unless either the SDK version is end of life or we rebuild the lambdas

(I think there's a workflow to rebuild all the lambdas, but personally I'd be afraid to run it for fear of breaking an existing pack)

Reply:
Yeah once this winds up in a built pack it becomes a pain to migrate. I'd prefer erring on the side of extensible shapes as well.

Reply:
I'm not sure we need another state here in the future and I don't think the added complexity in the handling path is worth it, but I'll do it anyways.